### ISSUE TITLE
[Steam Deck] Application cannot be used after SteamOS upgraded to 3.5.19

### ISSUE THREAD

**Issue by kaip97** [hexdump2002/game-miner-public#86]
After the Steam Deck upgraded to SteamOS 3.5.19 (build 20240422.1), the application can be started, but it keeps loading and unable to load the non-steam game list or make any changes.

**Comment by hexdump2002** [comment]
At this moment I'm into others projects and will take a bit of time for me to fix errors (PR's are very welcomed). But, in the meantime, would you mind executing the appImage from the terminal and tell me what error is it showing? To do this, you just have to run Konsole, go to the folder where the appImage is and execute it. Please, paste the output here or a screenshot of the error.

**Comment by kaip97** [comment]
Hi Hexdump, 

I had captured the Konsole output after running the appimage for 10mins, below error message for your reference.

Btw, thanks much for building such a great app, it had saved my time importing my own non-steam library.
```
(1)(deck@steamdeck ~)$ ./Desktop/GameMiner.AppImage 
Gtk-Message: 22:31:22.433: Failed to load module "canberra-gtk-module"
Gtk-Message: 22:31:22.483: Failed to load module "colorreload-gtk-module"
Gtk-Message: 22:31:22.484: Failed to load module "window-decorations-gtk-module"
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: RangeError (length): Invalid value: Not in inclusive range 0..13583185: 13583186
#0      BinaryBufferReader.readByte (package:game_miner/logic/io/binary_buffer_reader.dart)
#1      BinaryBufferReader.readUint16 (package:game_miner/logic/io/binary_buffer_reader.dart:168)
#2      CompatToolsDataProvider._isEndOfEntry (package:game_miner/data/data_providers/compat_tools_data_provider.dart:103)
#3      CompatToolsDataProvider._loadInternalCompatToolEntry (package:game_miner/data/data_providers/compat_tools_data_provider.dart:90)
#4      CompatToolsDataProvider.loadInternalCompatTools (package:game_miner/data/data_providers/compat_tools_data_provider.dart:66)
#5      CompatToolsRepository.loadCompatTools (package:game_miner/data/repositories/compat_tools_repository.dart:17)
<asynchronous suspension>

[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: RangeError (length): Invalid value: Not in inclusive range 0..13583185: 13583186
#0      BinaryBufferReader.readByte (package:game_miner/logic/io/binary_buffer_reader.dart)
#1      BinaryBufferReader.readUint16 (package:game_miner/logic/io/binary_buffer_reader.dart:168)
#2      CompatToolsDataProvider._isEndOfEntry (package:game_miner/data/data_providers/compat_tools_data_provider.dart:103)
#3      CompatToolsDataProvider._loadInternalCompatToolEntry (package:game_miner/data/data_providers/compat_tools_data_provider.dart:90)
#4      CompatToolsDataProvider.loadInternalCompatTools (package:game_miner/data/data_providers/compat_tools_data_provider.dart:66)
#5      CompatToolsRepository.loadCompatTools (package:game_miner/data/repositories/compat_tools_repository.dart:17)
<asynchronous suspension>
#6      GameMgrCubit._loadData (package:game_miner/logic/blocs/game_mgr_cubit.dart:157)
<asynchronous suspension>



```

**Comment by hexdump2002** [comment]
Sadly after checking the issue, I can't do much to fix it. It seems to be something about new Holo update on the deck not being totally compatible with the current flutter runtime. I will be vigilant on the issue and if the flutter team / valve fixes it I will upload a fix and release a new version.

**Comment by hexdump2002** [comment]
After having some talks with the flutter team and some folks at valve I have been able to get a patch for flutter and fix a bug produced by latest steam updates (jun 2024).

Take into account that Valve is making a lot of changes to Holo (more now that have teamed up with Arch team) and if they change their internal .vdf files for some files again... Things are likely to be broke again. This happened as I said past Jun with a lot of apps that modified and read those files.

I have sent a fixed AppImage to some folks on telegram and as soon as they give me green light I will release a new version here at github. So, the update popup should show in game miner for you to update.

On the other hand I have fixed some old bugs too. They weren't important but I did it anyway.

P.D. If you would like to test it prior release let me know.

Hope it works for you too.
